app/services/live_stream_updater.rb

```diff
@@ -0,0 +1,95 @@
+class LiveStreamUpdater
+  def initialize(object, state = nil)
+    @object = object
+    @state = state
+    @content_item = fetch_live_content_item
+  end
+
+  def updated?
+    update_object_and_content_item.try(:code) == 200
+  end
+
+  def published?
+    publish_content_item.try(:code) == 200
+  end
+
+  def resync
+    states_in_sync? ? object : object.toggle(:state)
+  end
+
+private
+
+  attr_reader :state, :live_stream, :content_item
+  attr_accessor :object
+
+  def states_in_sync?
+    live_state == object.state
+  end
+
+  def live_state
+    content_item["details"]["live_stream_enabled"]
+  end
+
+  def update_object_and_content_item
+    if object.update(state: state)
+      update_content_item
+    end
+  end
+
+  def update_content_item
+    with_longer_timeout do
+      begin
+        response =
+          Services.publishing_api.put_content(landing_page_id, live_stream_payload)
+        response.code == 200 ? response : object.toggle(:state)
```

Comment:
I think this line might confuse me, so I might have a think about how I might unconfuse myself